File: Python/S0049.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        group = {}
        for s in strs:
            count = [0] * 26
            for char in s:
                count[ord(char) - ord('a')] += 1
            key = tuple(count)
            if key not in group:
                group[key] = []
            group[key].append(s)
        return list(group.values())

File: C++/S0049.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> groups;
        for (string s : strs) {
            string key = string(26, 0);
            for (char c : s) key[c - 'a']++;
            groups[key].push_back(s);

        }
        vector<vector<string>> res;
        for (auto p : groups) res.push_back(p.second);
        return res;
    }
};